src/main/scala/org/bitcoins/core/gen/ScriptGenerators.scala

```diff
@@ -206,9 +277,49 @@ trait ScriptGenerators extends BitcoinSLogger {
     (p2SHScriptSignature, p2SHScriptPubKey, privateKeys)
   }
 
+  /**
+    * Generates a signed [[CLTVScriptSignature]] that spends from a [[CLTVScriptPubKey]] correctly
+    *
+    * @return the signed [[CLTVScriptSignature]], the [[CLTVScriptPubKey]] it spends, and the sequences of [[ECPrivateKey]]
+    *         used to sign the scriptSig
+    */
+  def signedCLTVScriptSignature(scriptNumber : ScriptNumber, lockTime : UInt32, sequence : UInt32) : Gen[(ScriptSignature, CLTVScriptPubKey, Seq[ECPrivateKey])] = for {
+    (cltv, privKey, num) <- cltvScriptPubKey(scriptNumber)
+  } yield {
+    val (creditingTx, outputIndex) = TransactionGenerators.buildCreditingTransaction(cltv)
+    val scriptSig = CLTVScriptSignature(cltv, Seq(EmptyDigitalSignature), Seq(privKey.publicKey), None)
+    val (unsignedSpendingTx, inputIndex) = {
+      TransactionGenerators.buildSpendingTransaction(TransactionConstants.version, creditingTx, scriptSig, outputIndex, lockTime, sequence)
+    }
+    val txSignatureComponent = TransactionSignatureComponent(unsignedSpendingTx, inputIndex, cltv, Policy.standardScriptVerifyFlags)
+    val txSignature = TransactionSignatureCreator.createSig(txSignatureComponent, privKey, SIGHASH_ALL.defaultValue)
+    val signedScriptSig = CLTVScriptSignature(cltv, Seq(txSignature), Seq(privKey.publicKey), None)
+    (signedScriptSig, cltv, Seq(privKey))
+  }
+
+  /**
+    * Generates a signed [[CSVScriptSignature]] that spends from a [[CSVScriptPubKey]] correctly
+    * @return the signed [[CSVScriptSignature]], the [[CSVScriptPubKey]] it spends, and the sequences of [[ECPrivateKey]]
+    *         used to sign the scriptSig
+    */
+  def signedCSVScriptSignature(csvScriptNum : ScriptNumber, sequence : UInt32) : Gen[(ScriptSignature, CSVScriptPubKey, Seq[ECPrivateKey])] = for {
```

Comment:
I think this should technically return a `CSVScriptSignature` instead of a generic `ScriptSignature`